packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -664,17 +666,9 @@ private DocumentAsset uploadPDFFileToDocumentThenExtract(final String doi, final
 			}
 
 			// fire and forgot pdf extractions
-			final ResponseEntity<DocumentAsset> res = knowledgeController.postPDFToCosmos(docId);
-			if (res.getStatusCode().is2xxSuccessful()) {
-				throw new ResponseStatusException(
-						org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
-						"Unable to extract pdf");
-			}
-			return res.getBody();
-
+			extractionService.extractPDF(docId, currentUserId);
 		} catch (final Exception e) {
 			log.error("Unable to upload PDF document then extract", e);
```

Comment:
If this fails no exception is passed on to the front end. Should this be changed to returning a boolean for success that the upstream use can check?

Reply:
The front-end should be subscribed to EXTRACTION messages - as this function takes time - no need to waste the UIs time - new workflow Yohann and I are trying out